### PR TITLE
Feature/smk plugin

### PR DESCRIPTION
**Pull request by wshands**
Plugin for supporting Snakemake workflows intended as a start for our collaborators. The bulk of functionality will be added by collaborators.
Addresses https://ucsc-cgl.atlassian.net/browse/SEAB-3752

### CODE REVIEW

**Review by garyluu — APPROVED** [review]
Assuming the collaborators are fine with working on Java and keeping it up to date.

**Review by coverbeck — COMMENTED** [review]
Before reviewing the content of this PR, it seems odd to me that this repo is forked from the galaxy plugin. It's not like we're ever going to push changes back to the Galaxy repo. Not the end of the world, but it also makes the PR diff a little jarring.

**Comment by wshands** [comment]
Yes I would like to make a copy of this repo maybe after the PR as outlined here https://stackoverflow.com/questions/16052477/delete-fork-dependency-of-a-github-repository

**Review by coverbeck — CHANGES_REQUESTED** [review]
I tried building this and it fails:

```
mvn clean install
[INFO] Scanning for projects...
[INFO]
[INFO] --< com.github.dockstore.dockstore-snakemake-interface:dockstore-snakemake-interface >--
[INFO] Building dockstore-snakemake-interface 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[WARNING] The POM for io.dockstore:dockstore-language-plugin-parent:jar:1.12.0-alpha.2-SNAPSHOT is missing, no dependency information available
[WARNING] The POM for io.dockstore:dockstore-common:jar:1.12.0-alpha.2-SNAPSHOT is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.426 s
[INFO] Finished at: 2022-01-31T15:51:28-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project dockstore-snakemake-interface: Could not resolve dependencies for project com.github.dockstore.dockstore-snakemake-interface:dockstore-snakemake-interface:jar:0.0.1-SNAPSHOT: The following artifacts could not be resolved: io.dockstore:dockstore-language-plugin-parent:jar:1.12.0-alpha.2-SNAPSHOT, io.dockstore:dockstore-common:jar:1.12.0-alpha.2-SNAPSHOT: Could not find artifact io.dockstore:dockstore-language-plugin-parent:jar:1.12.0-alpha.2-SNAPSHOT -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
```

**Review by denis-yuen — APPROVED** [review]
Seems like a good start,
Might want to demo harvesting something from `.snakemake-workflow-catalog.yml` like launch-with WIP instructions  but can be follow-up ticket

File: pom.xml [hunk 1]
```diff
@@ -2,21 +2,22 @@
...
-    <name>dockstore-galaxy-interface</name>
-    <version>0.0.7-SNAPSHOT</version>
+    <name>dockstore-snakemake-interface</name>
+    <version>0.0.1-SNAPSHOT</version>
 
     <properties>
-        <github.url>scm:git:git@github.com:galaxyproject/dockstore-galaxy-interface.git</github.url>
-        <dockstore.version>1.9.0-alpha.1</dockstore.version>
+        <github.url>scm:git:git@github.com:dockstore/dockstore-snakemake-interface.git</github.url>
+<!--        <dockstore.version>1.9.0-alpha.1</dockstore.version>-->
```
[the remark this answers is not in the document]
Comment (wshands) [reply pom.xml]: Set to dockstore development version until a dockstore release with Snakemake support

**Review by coverbeck — CHANGES_REQUESTED** [review]
This still doesn't build for me, although with a different error than before:

```
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /Users/charlesoverbeck/git/dockstore/dockstore-snakemake-interface/src/main/java/org/dockstore/dockstore_snakemake_interface/language/SnakeMakeWorkflowPlugin.java:[120,32] cannot find symbol
  symbol:   variable SMK
  location: class io.dockstore.common.DescriptorLanguage
[INFO] 1 error
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.028 s
[INFO] Finished at: 2022-02-24T09:19:02-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project dockstore-snakemake-interface: Compilation failure
[ERROR] /Users/charlesoverbeck/git/dockstore/dockstore-snakemake-interface/src/main/java/org/dockstore/dockstore_snakemake_interface/language/SnakeMakeWorkflowPlugin.java:[120,32] cannot find symbol
[ERROR]   symbol:   variable SMK
[ERROR]   location: class io.dockstore.common.DescriptorLanguage
```

I know this is the first iteration where we are just getting things set up, that it's not being released yet, so if you want to push back and merge this regardless, I'm OK with that. but since you asked for a review and since it doesn't build, I'm going to request changes.

**Comment by wshands** [comment]
> This still doesn't build for me, although with a different error than before:
>
> I know this is the first iteration where we are just getting things set up, that it's not being released yet, so if you want to push back and merge this regardless, I'm OK with that. but since you asked for a review and since it doesn't build, I'm going to request changes.

Added SNAPSHOT from Artifactory

**Review by coverbeck — APPROVED** [review]
Now builds for me!

**Pull request merged**